add note about arguments in shebang in modules (#1718)
ansible/ansible-documentation: docs/docsite/rst/dev_guide/developing_modules_documenting.rst

Begin your Ansible module with ``#!/usr/bin/python`` - this "shebang" allows ``ansible_python_interpreter`` to work. Follow the shebang immediately with ``# -*- coding: utf-8 -*-`` to clarify that the file is UTF-8 encoded.

> .. warning:: Using ``#!/usr/bin/env``, makes ``env`` the interpreter and bypasses ``ansible_<interpreter>_interpreter`` logic. Using arguments in combination with ``env`` (e.g. ``#!/usr/bin/env python``) does not work.

Reviewer: It is not just env, it is with any interpreter, i would make this a separate warning. 
```
.. warning:: 
  - Using ``#!/usr/bin/env``, makes ``env`` the interpreter and bypasses ``ansible_<interpreter>_interpreter`` logic.    
  -  Passing arguments to the interpreter in the shebang also does not work.
```

Author: Thanks. I've updated it.

Begin your Ansible module with ``#!/usr/bin/python`` - this "shebang" allows ``ansible_python_interpreter`` to work. Follow the shebang immediately with ``# -*- coding: utf-8 -*-`` to clarify that the file is UTF-8 encoded.

> .. warning:: Using ``#!/usr/bin/env``, makes ``env`` the interpreter and bypasses ``ansible_<interpreter>_interpreter`` logic.
> .. warning:: Passing arguments to the interpreter (e.g. ``#!/usr/bin/env python``) in the shebang also does not work.

Reviewer 2: ```suggestion
.. warning::
   - Using ``#!/usr/bin/env`` makes ``env`` the interpreter and bypasses ``ansible_<interpreter>_interpreter`` logic.
   - Passing arguments to the interpreter in the shebang does not work (for example, ``#!/usr/bin/env python``) .
```
